src/websocket/common.ts

```diff
@@ -5,3 +5,26 @@ export const catchError = (res: any) => {
     ? Promise.reject(res)
     : Promise.resolve(res)
 }
+
+export const promiseTimeout = (ms: number, promise: Promise<any>) => {
+  let id
+  let timeout = new Promise((resolve, reject) => {
+    id = setTimeout(() => {
+      clearTimeout(id);
+      reject(new Error('Timed out in '+ ms + 'ms.'))
+    }, ms)
+  })
+
+  return new Promise((res, rej) => {
+    Promise.race([
+      promise,
+      timeout
+    ]).then((a) => {
+      clearTimeout(id);
+      return res(a)
+    })
+    .catch(e => {
+      return rej(new Error(e))
```

Comment:
```suggestion
      return rej(e)
```
I think doing this error forwarding you want to forward it plain, not re-wrapped in an Error since it already should be